javascript/javascript3/week1/script.js

```diff
@@ -0,0 +1,143 @@
+        this.backgroundChange(main);
+        document.querySelector(".weather").classList.remove("loading");
+    },
+    search: function () {
+        if (!document.querySelector(".search-bar").value) {
+            alert("Enter a valid city name");
+            return
+        } else {
+            this.fetchWeather(document.querySelector(".search-bar").value);
+        }
+    },
+    backgroundChange: function (main) {
+        let backgroundHolder = document.querySelector(".card");
+        if (main == "Thunderstorm") {
+            backgroundHolder.style.backgroundImage = "url('gifs/thunderstorm.gif')";
+        } else if (main == "Drizzle") {
+            backgroundHolderstyle.backgroundImage = "url('gifs/drizzle.gif')";
+        } else if (main == "Rain") {
+            backgroundHolder.style.backgroundImage = "url('gifs/rain.gif')";
+        } else if (main == "Snow") {
+            backgroundHolderstyle.backgroundImage = "url('gifs/snow.gif')";
+        } else if (main == "Mist") {
+            backgroundHolder.style.backgroundImage = "url('gifs/mist.gif')";
+        } else if (main == "Smoke") {
+            backgroundHolder.style.backgroundImage = "url('gifs/mist.gif')";
+        } else if (main == "Haze") {
+            backgroundHolder.style.backgroundImage = "url('gifs/mist.gif')";
+        } else if (main == "Dust") {
+            backgroundHolder.style.backgroundImage = "url('gifs/dust.gif')";
+        } else if (main == "Fog") {
+            backgroundHolder.style.backgroundImage = "url('gifs/fog.gif')";
+        } else if (main == "Ash") {
+            backgroundHolder.style.backgroundImage = "url('gifs/ash.gif')";
+        } else if (main == "Squall") {
+            backgroundHolder.style.backgroundImage = "url('gifs/tornado.gif')";
+        } else if (main == "Tornado") {
+            backgroundHolder.style.backgroundImage = "url('gifs/tornado.gif')";
+        } else if (main == "Clear") {
+            backgroundHolder.style.backgroundImage = "url('gifs/clear.gif')";
+        } else if (main == "Clouds") {
+            backgroundHolder.style.backgroundImage = "url('gifs/cloud.gif')";
+        }
+    },
+    timeConverter: function (unix) {
+        const date = new Date(unix * 1000);
+        const hours = date.getHours();
+        const minutes = "0" + date.getMinutes();
+        const formattedTime = hours + ':' + minutes.substr(-2);
+        return formattedTime;
+    },
+};
+
+weather.fetchWeather(localStorage.name);
+document.querySelector(".search .searchButton")
+    .addEventListener("click", function () {
+        weather.search();
+    });
+document.querySelector(".search-bar").addEventListener("keyup", function (event) {
+    if (event.key == "Enter") {
+        weather.search()
+    } else if (event.key == "+") {
+        weather.searchWithJson()
+    }
+});
+document.querySelector(".search .locationButton")
+    .addEventListener("click", function () {
```

Comment:
I believe you can just set the function to be weather.fetchWeatherWithLocation, and it should get called